force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +122,28 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
+        Integer vowelCount = 0;
 
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+        // Define a set containing lowercase vowels as strings
+        Set<String> vowels = new Set<String>{'a', 'e', 'i', 'o', 'u'};
+        
+        // Convert the input string to lowercase to handle case-insensitivity
+        str = str.toLowerCase();
+        
+        // Loop through each character in the string
+        for (Integer i = 0; i < str.length(); i++) {
+            // Get the current character
+            String letter = str.substring(i, i+1);
+            
+            // Check if the character is a vowel
+            if (vowels.contains(letter)) {
+                // If it is, increment the vowel count
+                vowelCount++;
+            }
         }
-        */
```

Comment:
This is an interesting solutions and not one I thought of. I would have to bench mark it. But I think calling String.split would be more efficient. In this case you are generating a new String on each iteration and putting it in memory vs doing it once in a list. 

My solutions is below. 

```
  public static Integer countVowels(String str) {
        // Initialize the result Integer
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
    ```